sdk/servicebus/Azure.Messaging.ServiceBus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,309 @@
+
+  - **Service Timeout** : This indicates that the Service Bus service did not respond to an operation within the expected amount of time. This may have been caused by a transient network issue or service problem. The Service Bus service may or may not have successfully completed the request; the status is not known. In the case of accepting the next available session, this exception indicates that there were no unlocked sessions available in the entity.
+
+  - **Quota Exceeded** : This typically indicates that there are too many active receiver operations for a single entity. In order to avoid this error, reduce the number of potential concurrent receives. You can use batch receives to attempt to receive multiple messages per receive request. Please see [Service Bus quotas][ServiceBusQuotas] for more information.
+
+  - **Message Size Exceeded** : This indicates that the max message size has been exceeded. For standard tier namespaces, a single message may not exceed 256KB. For premium tier namespaces, the max message size is 100MB. This includes the body of the message, as well as any associated metadata and system overhead. The best approach for resolving this error is to reduce the number of messages being sent in a batch or the size of the body included in the message. Because size limits are subject to change, please refer to [Service Bus quotas][ServiceBusQuotas] for specifics.  
+  
+  - **MessageLockLost** : This indicates that the lock on the message is lost. Callers should call attempt to receive and process the message again. This only applies to non-session entities. This error occurs if processing takes longer than the lock duration and the message lock is not renewed. Note that this error can also occur when the link is detached due to a transient network issue or when the link is idle for 10 minutes.
+  
+  - **SessionLockLost**: This indicates that the lock on the session has expired. Callers should attempt to accept the session again. This only applies to session-enabled entities. This error occurs if processing takes longer than the lock duration and the session lock is not renewed. Note that this error can also occur when the link is detached due to a transient network issue or when the link is idle for 10 minutes.
+
+  - **SessionCannotBeLocked**: This indicates that the requested session cannot be locked because the lock is already held elsewhere. Once the lock expires, the session can be accepted.
+
+### Other common exceptions
+
+- **ArgumentException** : An exception deriving from `ArgumentException` is thrown by clients when a parameter provided when interacting with the client is invalid. Information about the specific parameter and the nature of the problem can be found in the `Message`.
```

Comment:
These exceptions don't seem like they're worth documenting, mostly because the _text_ of the error message for all of them should tell you what your next step is, unlike the others which might not be as intuitive.

Reply:
You'd be surprised at the feedback/questions.   There's value in having these represented for the support teams.

Reply:
Next time it happens it'd be interesting to see if the error message is somehow unclear. 

On the other hand, I do recall us having some issues with people noticing the actual error because it can sometimes become buried under the callstack. So I see your point. :)